doc/bql.rst

```diff
@@ -446,16 +447,15 @@ subquery instead:
 
 .. index:: ``SIMILARITY``
 
-``SIMILARITY [TO (<expression>)] IN THE CONTEXT OF <column>``
-
-   Function of one or two implied rows.  If given ``TO``, returns a
-   measure of the similarity of the implied row with the first row
-   satisfying <expression>.  Otherwise, returns a measure of the
-   similarity of the two implied rows.  The similarity may be
-   considered with in the context of a column.
+``SIMILARITY [OF (<expression0>)] [TO (<expression1>)] IN THE CONTEXT OF <column>``
 
-   *Columns* is a comma-separated list of column names or
-   ``ESTIMATE * FROM COLUMNS OF ...`` subqueries.
+   Constant, or function of one or two implied rows. If given both ``OF`` and
+   ``TO``, returns a constant measure of similarity between the first row
+   satisfied by *expression0* and the first row satisfied by *expression1*. If
+   given only  ``TO`` returns a measure of the similarity of the implied row
+   with the first row satisfying *expression1*. Otherwise, returns a measure of
+   the similarity of the two implied rows.  The similarity may be
+   considered within the context of a column.
```

Comment:
The terminology that I'm calling awkward is specifically 'function of an implied row' or 'function of two implied columns', not the words 'function' and 'constant'.  Maybe a better choice would be 'a 2-row function', or 'this expression works in a 2-row context', or something like that.  I don't have any brilliant ideas for how to replace it at the moment -- hence not making a concrete suggestion here, just ruminating.